py/server/deephaven/pandas.py

```diff
@@ -43,7 +45,25 @@ def _column_to_series(table: Table, col_def: Column) -> pandas.Series:
         raise DHError(e, message="failed to create apandas Series for {col}") from e
 
 
-def to_pandas(table: Table, cols: List[str] = None) -> pandas.DataFrame:
+_PA_PD_DTYPE_MAPPING = {
+    pa.int8(): pd.ArrowDtype(pa.int8()),
+    pa.int16(): pd.ArrowDtype(pa.int16()),
+    pa.int32(): pd.ArrowDtype(pa.int32()),
+    pa.int64(): pd.ArrowDtype(pa.int64()),
+    pa.uint8(): pd.ArrowDtype(pa.uint16()),
+    pa.uint16(): pd.ArrowDtype(pa.uint64()),
+    pa.uint32(): pd.ArrowDtype(pa.uint32()),
+    pa.uint64(): pd.ArrowDtype(pa.uint64()),
+    pa.bool_(): pd.ArrowDtype(pa.bool_()),
+    pa.float32(): pd.ArrowDtype(pa.float32()),
+    pa.float64(): pd.ArrowDtype(pa.float64()),
+    pa.string(): pd.ArrowDtype(pa.string()),
+    pa.timestamp('ns'): pd.ArrowDtype(pa.timestamp('ns')),
+    pa.timestamp('ns', tz='UTC'): pd.ArrowDtype(pa.timestamp('ns', tz='UTC')),
+}
+
+
+def to_pandas(table: Table, cols: List[str] = None, pyarrow_backed: bool = False) -> pandas.DataFrame:
```

Comment:
Some questions on this after some more thought and docs reading https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html.

1) Should we use the `pyarrow_backend: bool` variable or `dtype_backend: str` like pandas uses, for consistency and flexability?
2) `read_csv` is defaulting to `numpy_nullable`.  Do we want that default for consistency or "pyarrow"  for performance and a lean to the future instead of the past?

Reply:
1. Although what we do resembles much more to pyarrow.Table.to_pandas(), I do see the benefits of being consistent with what Pandas uses.
2. read_csv, read_parquet etc. actually default to the Numpy backend without nullable dtypes. This is getting confusing because Pandas now supports 3 backends: no-nullable dtype Numpy arrays (eg. int8), nullable dtype Numpy arrays (eg. Int8), and arrow-types (which are always nullable).